ui/selectors/multichain.ts

```diff
@@ -0,0 +1,154 @@
+import { isEvmAccountType } from '@metamask/keyring-api';
+import { ProviderConfig } from '@metamask/network-controller';
+import {
+  CaipChainId,
+  KnownCaipNamespace,
+  parseCaipChainId,
+} from '@metamask/utils';
+import {
+  MultichainProviderConfig,
+  MULTICHAIN_PROVIDER_CONFIGS,
+} from '../../shared/constants/multichain/networks';
+import {
+  getNativeCurrency,
+  getProviderConfig,
+} from '../ducks/metamask/metamask';
+import { AccountsState } from './accounts';
+import {
+  getAllNetworks,
+  getCurrentCurrency,
+  getNativeCurrencyImage,
+  getSelectedInternalAccount,
+  getShouldShowFiat,
+} from '.';
+
+export type MultichainState = AccountsState & {
+  metamask: {
+    // TODO: Use states from new {Rates,Balances,Chain}Controller
+  };
+};
+
+export type MultichainNetwork = {
+  nickname: string;
+  isEvmNetwork: boolean;
+  chainId?: CaipChainId;
+  network?: ProviderConfig | MultichainProviderConfig;
+};
+
+export function getMultichainNetworkProviders(
+  _state: MultichainState,
+): MultichainProviderConfig[] {
+  // TODO: need state from the ChainController?
+  return Object.values(MULTICHAIN_PROVIDER_CONFIGS);
+}
+
+export function getMultichainNetwork(
+  state: MultichainState,
+): MultichainNetwork {
+  const selectedAccount = getSelectedInternalAccount(state);
+  const isEvm = isEvmAccountType(selectedAccount.type);
+
+  // EVM networks
+  const evmNetworks: ProviderConfig[] = getAllNetworks(state);
+  const evmProvider: ProviderConfig = getProviderConfig(state);
+
+  if (isEvm) {
+    const evmChainId =
+      `${KnownCaipNamespace.Eip155}:${evmProvider.chainId}` as CaipChainId;
+    const evmNetwork = evmNetworks.find(
+      (network) => network.chainId === evmProvider.chainId,
+    );
+
+    return {
+      nickname: 'Ethereum',
```

Comment:
We should get the nickname fro mthe evm provider if it is available. If its not maybe nickname should be its chain id instead of Ethereum?

Reply:
We'll address this later on other UI PRs!